service/consul/environment.go

```diff
@@ -69,9 +69,6 @@ func ensureIDs(r *api.AgentServiceRegistration) {
 
 func newInstancerKey(w Watch) string {
 	datacenter := w.QueryOptions.Datacenter
```

Comment:
we should remove this and put `w.QueryOptions.Datacenter` directly into the `fmt.Sprintf` call below